Unstable CI: `TestRestoringFromDifferentHeightsWithFullHistory`

# Task Overview

The test `TestRestoringFromDifferentHeightsWithFullHistory` has been seen to sometimes fail in three separate ways, the first being (`TestRestoreFromFullHistorySnapshotAndProcessEvents` has also been seen to fail this way):
```
[2022-11-10T14:04:25.289Z] === RUN   TestRestoringFromDifferentHeightsWithFullHistory
[2022-11-10T14:04:25.289Z] 2022-11-10T13:54:01.368Z	INFO	snapshot schema creation	v3@v3.6.1/migration.go:38	OK   0001_initial.sql
[2022-11-10T14:04:25.289Z] 2022-11-10T13:54:01.382Z	INFO	snapshot schema creation	v3@v3.6.1/up.go:210	goose: no migrations to run. current version: 1
[2022-11-10T14:04:25.289Z] 2022-11-10T13:54:01.382Z	INFO	root	dehistory/service_test.go:447	fetching history for segment id:QmNdrVHMKW1KtyNJeZteHZJ1ka3ejY1rtT66mW3EGTWGVw
[2022-11-10T14:04:25.289Z] 2022-11-10T13:54:01.388Z	INFO	root	dehistory/service_test.go:447	fetched history:{HeightFrom:1 HeightTo:1000 PreviousHistorySegmentID:}
[2022-11-10T14:04:25.289Z] 2022-11-10T13:54:01.418Z	INFO	root	dehistory/service_test.go:287	creating database
[2022-11-10T14:04:25.289Z]     service_test.go:288: 
[2022-11-10T14:04:25.289Z]         	Error Trace:	/jenkins/workspace/vega_develop/vega/datanode/dehistory/service_test.go:288
[2022-11-10T14:04:25.289Z]         	Error:      	Received unexpected error:
[2022-11-10T14:04:25.289Z]         	            	failed to create vega database: failed to drop database:unable to drop existing database:conn closed
[2022-11-10T14:04:25.289Z]         	Test:       	TestRestoringFromDifferentHeightsWithFullHistory
[2022-11-10T14:04:25.289Z] --- FAIL: TestRestoringFromDifferentHeightsWithFullHistory (50.69s)
```

the second:
```
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:14.415Z	INFO	snapshot schema creation	v3@v3.6.1/migration.go:38	OK   0001_initial.sql
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:14.419Z	INFO	snapshot schema creation	v3@v3.6.1/up.go:210	goose: no migrations to run. current version: 1
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:14.420Z	INFO	root	dehistory/service_test.go:447	fetching history for segment id:QmdUrVBM71aSp8jqnAh3ttS59zGXBFstog64u3VdnC776F
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:14.437Z	INFO	root	dehistory/service_test.go:447	fetched history:{HeightFrom:2001 HeightTo:3000 PreviousHistorySegmentID:QmbG5YMPbhnVQUyaFXsVMwTWbbBoL88ahTuAVLTYr5jJAJ}
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:14.498Z	INFO	root	dehistory/service_test.go:287	creating database
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:15.002Z	INFO	root	dehistory/service_test.go:287	creating schema
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:18.997Z	INFO	snapshot schema creation	v3@v3.6.1/migration.go:38	OK   0001_initial.sql
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:19.005Z	INFO	snapshot schema creation	v3@v3.6.1/up.go:210	goose: no migrations to run. current version: 1
[2022-11-08T18:32:17.654Z] 2022-11-08T18:22:19.151Z	INFO	root	dehistory/service.go:174	preparing for bulk load
[2022-11-08T18:32:17.654Z]     service_test.go:288: 
[2022-11-08T18:32:17.654Z]         	Error Trace:	/jenkins/workspace/vega_master/vega/datanode/dehistory/service_test.go:288
[2022-11-08T18:32:17.654Z]         	Error:      	Received unexpected error:
[2022-11-08T18:32:17.654Z]         	            	failed to load snapshot data:failed to prepare database for bulk load: failed to execute drop constrain ALTER TABLE public.reward_scores DROP CONSTRAINT reward_scores_pkey;: unexpected EOF
[2022-11-08T18:32:17.654Z]         	Test:       	TestRestoringFromDifferentHeightsWithFullHistory
[2022-11-08T18:32:17.654Z] --- FAIL: TestRestoringFromDifferentHeightsWithFullHistory (42.17s)
```
the third:
```
[2022-11-10T12:24:06.692Z] 2022-11-10T12:16:43.185Z	INFO	root	snapshot/service_load_snapshot.go:54	copying data into database	{"database version": 1}
[2022-11-10T12:24:06.692Z] 2022-11-10T12:16:43.286Z	INFO	root	snapshot/service_load_snapshot.go:58	copying testnet-1-1000-historysnapshot into database
[2022-11-10T12:24:06.692Z]     service_test.go:288: 
[2022-11-10T12:24:06.692Z]         	Error Trace:	/jenkins/workspace/vega_PR-6774/vega/datanode/dehistory/service_test.go:288
[2022-11-10T12:24:06.692Z]         	Error:      	Received unexpected error:
[2022-11-10T12:24:06.692Z]         	            	failed to load snapshot data:failed to load history snapshot {History Snapshot for Chain ID:testnet Height From:1 Height To:1000}: failed to copy uncompressed data into the database testnet-1-1000-historysnapshot : failed to disable triggers, setting session replication role to replica failed:write failed: write unix @->/tmp/84f41c65-f7f3-4c69-9b15-37b5c835799b3005738789/sqlstore/.s.PGSQL.5432: write: broken pipe
[2022-11-10T12:24:06.692Z]         	Test:       	TestRestoringFromDifferentHeightsWithFullHistory
[2022-11-10T12:24:06.692Z] --- FAIL: TestRestoringFromDifferentHeightsWithFullHistory (5.90s)
```
## Specs
- [Link](xyz) to spec or milestone document info for the feature

# Acceptance Criteria
How do we know when this technical task is complete:
- It is possible to...
- Vega is able to...

# Test Scenarios
Detailed scenarios (1-3!) that can be executed as feature tests to verify that the feature has been implemented as expected.

GIVEN (setup/context) 
WHEN (action) 
THEN (assertion) 
See [here](https://github.com/vegaprotocol/vega/tree/develop/integration/) for more format information and examples.

# Dependencies
Links to any tickets that have a dependant relationship witht his task.

# Additional Details (optional)
Any additional information including known dependencies, impacted components.

# Examples (optional)
Code snippets, API calls that could be used on dependant tasks.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Create relevant for [system-test](https://github.com/vegaprotocol/system-tests/issues) tickets with feature labels
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All acceptance criteria confirmed to be met, or, reasons why not discussed with the engineering leadership team
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new system-tests
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -48,6 +48,8 @@
 - [6767](https://github.com/vegaprotocol/vega/issues/6767) - Protocol upgrade rejected events fail to write in the database
 - [6896](https://github.com/vegaprotocol/vega/issues/6896) - Fix timestamps in proposals (`GQL`)
 - [6844](https://github.com/vegaprotocol/vega/issues/6844) - Use proper type in `GQL` for transfer types and some types rename
+- [6783](https://github.com/vegaprotocol/vega/issues/6783) - Unstable `CI` tests for `dehistory`
+- [6844](https://github.com/vegaprotocol/vega/issues/6844) - Unstable `CI` tests for `dehistory`
 
 
 ## 0.62.1

--- a/cmd/data-node/commands/dehistory/load.go
+++ b/cmd/data-node/commands/dehistory/load.go
@@ -68,7 +68,7 @@ func (cmd *loadCmd) Execute(_ []string) error {
 	snapshotService, err := snapshot.NewSnapshotService(log, cmd.Config.DeHistory.Snapshot, cmd.Config.SQLStore.ConnectionConfig,
 		vegaPaths.StatePathFor(paths.DataNodeDeHistorySnapshotCopyFrom),
 		vegaPaths.StatePathFor(paths.DataNodeDeHistorySnapshotCopyTo), func(version int64) error {
-			if err = sqlstore.MigrateToSchemaVersion(log, cmd.Config.SQLStore, version); err != nil {
+			if err = sqlstore.MigrateToSchemaVersion(log, cmd.Config.SQLStore, version, sqlstore.EmbedMigrations); err != nil {
 				return fmt.Errorf("failed to migrate to schema version %d: %w", version, err)
 			}
 			return nil
@@ -116,7 +116,7 @@ func (cmd *loadCmd) Execute(_ []string) error {
 	if yes {
 		fmt.Printf("Loading history from block %d to %d...\n", from, to)
 
-		loadedFrom, loadedTo, err := deHistoryService.LoadAllAvailableHistoryIntoDatanode(context.Background())
+		loadedFrom, loadedTo, err := deHistoryService.LoadAllAvailableHistoryIntoDatanode(context.Background(), sqlstore.EmbedMigrations)
 		if err != nil {
 			return fmt.Errorf("failed to load all available history:%w", err)
 		}

--- a/cmd/data-node/commands/start/node_pre.go
+++ b/cmd/data-node/commands/start/node_pre.go
@@ -106,7 +106,7 @@ func (l *NodeCommand) persistentPre([]string) (err error) {
 	}
 
 	if dataNodeHasData && bool(l.conf.SQLStore.WipeOnStartup) {
-		if err = sqlstore.WipeDatabase(l.Log, l.conf.SQLStore.ConnectionConfig); err != nil {
+		if err = sqlstore.WipeDatabase(l.Log, l.conf.SQLStore.ConnectionConfig, sqlstore.EmbedMigrations); err != nil {
 			return fmt.Errorf("failed to wiped database:%w", err)
 		}
 		dataNodeHasData = false
@@ -287,7 +287,7 @@ func (l *NodeCommand) initialiseDecentralizedHistory() error {
 	l.snapshotService, err = snapshot.NewSnapshotService(snapshotServiceLog, l.conf.DeHistory.Snapshot,
 		l.conf.SQLStore.ConnectionConfig, l.vegaPaths.StatePathFor(paths.DataNodeDeHistorySnapshotCopyFrom),
 		l.vegaPaths.StatePathFor(paths.DataNodeDeHistorySnapshotCopyTo), func(version int64) error {
-			if err = sqlstore.MigrateToSchemaVersion(deHistoryLog, l.conf.SQLStore, version); err != nil {
+			if err = sqlstore.MigrateToSchemaVersion(deHistoryLog, l.conf.SQLStore, version, sqlstore.EmbedMigrations); err != nil {
 				return fmt.Errorf("failed to migrate to schema version %d: %w", version, err)
 			}
 			return nil

--- a/datanode/dehistory/initialise/initialise.go
+++ b/datanode/dehistory/initialise/initialise.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"io/fs"
 	"math/rand"
 	"net"
 	"sort"
@@ -29,7 +30,7 @@ var ErrChainNotFound = errors.New("no chain found")
 type deHistoryService interface {
 	GetActivePeerAddresses() []string
 	FetchHistorySegment(ctx context.Context, historySegmentID string) (store.SegmentIndexEntry, error)
-	LoadAllAvailableHistoryIntoDatanode(ctx context.Context) (loadedFrom int64, loadedTo int64, err error)
+	LoadAllAvailableHistoryIntoDatanode(ctx context.Context, sqlFs fs.FS) (loadedFrom int64, loadedTo int64, err error)
 }
 
 func DatanodeFromDeHistory(parentCtx context.Context, cfg Config, log *logging.Logger,
@@ -87,7 +88,7 @@ func DatanodeFromDeHistory(parentCtx context.Context, cfg Config, log *logging.L
 	log.Infof("fetched %d blocks from decentralised history", blocksFetched)
 
 	log.Infof("loading history into the datanode")
-	from, to, err := deHistoryService.LoadAllAvailableHistoryIntoDatanode(ctx)
+	from, to, err := deHistoryService.LoadAllAvailableHistoryIntoDatanode(ctx, sqlstore.EmbedMigrations)
 	if err != nil {
 		return fmt.Errorf("failed to load history into the datanode%w", err)
 	}

--- a/datanode/dehistory/service.go
+++ b/datanode/dehistory/service.go
@@ -158,7 +158,7 @@ func (d *Service) GetActivePeerAddresses() []string {
 	return activePeerIPAddresses
 }
 
-func (d *Service) LoadAllAvailableHistoryIntoDatanode(ctx context.Context) (loadedFrom int64, loadedTo int64, err error) {
+func (d *Service) LoadAllAvailableHistoryIntoDatanode(ctx context.Context, sqlFs fs.FS) (loadedFrom int64, loadedTo int64, err error) {
 	defer func() { _ = fsutil.RemoveAllFromDirectoryIfExists(d.snapshotsCopyFromDir) }()
 
 	err = os.MkdirAll(d.snapshotsCopyFromDir, fs.ModePerm)
@@ -183,7 +183,7 @@ func (d *Service) LoadAllAvailableHistoryIntoDatanode(ctx context.Context) (load
 	}
 
 	d.log.Infof("creating database")
-	if err = sqlstore.RecreateVegaDatabase(ctx, d.log, d.connConfig); err != nil {
+	if err = sqlstore.WipeDatabase(d.log, d.connConfig, sqlFs); err != nil {
 		return 0, 0, fmt.Errorf("failed to create vega database: %w", err)
 	}
 

--- a/datanode/dehistory/service_test.go
+++ b/datanode/dehistory/service_test.go
@@ -13,6 +13,7 @@ import (
 	"path/filepath"
 	"strconv"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
@@ -467,17 +468,22 @@ func migrateDatabase(version int64) error {
 func newTestEventSourceWithProtocolUpdateMessage() *TestEventSource {
 	var err error
 	var currentBlock *entities.Block
+	var m sync.RWMutex
 	evtSource, err := newTestEventSource(func(e events.Event, evtsCh chan<- events.Event) {
 		switch e.Type() {
 		case events.EndBlockEvent:
 			if currentBlock != nil && currentBlock.Height == 2500 {
+				m.RLock()
 				evtsCh <- events.NewProtocolUpgradeStarted(context.Background(), eventsv1.ProtocolUpgradeStarted{
 					LastBlockHeight: uint64(currentBlock.Height),
 				})
+				m.RUnlock()
 			}
 		case events.BeginBlockEvent:
 			beginBlock := e.(entities.BeginBlockEvent)
+			m.Lock()
 			currentBlock, err = entities.BlockFromBeginBlock(beginBlock)
+			m.Unlock()
 			if err != nil {
 				panic(err)
 			}

--- a/datanode/dehistory/service_test.go
+++ b/datanode/dehistory/service_test.go
@@ -480,8 +480,8 @@ func newTestEventSourceWithProtocolUpdateMessage() *TestEventSource {
 				m.RUnlock()
 			}
 		case events.BeginBlockEvent:
-			beginBlock := e.(entities.BeginBlockEvent)
 			m.Lock()
+			beginBlock := e.(entities.BeginBlockEvent)
 			currentBlock, err = entities.BlockFromBeginBlock(beginBlock)
 			m.Unlock()
 			if err != nil {

--- a/datanode/dehistory/service_test.go
+++ b/datanode/dehistory/service_test.go
@@ -76,13 +76,14 @@ var (
 
 	testMigrationsDir       string
 	testMigrationVersionNum int
+	sqlFs                   fs.FS
 )
 
 func TestMain(t *testing.M) {
 	outerCtx, cancelOuterCtx := context.WithCancel(context.Background())
 	defer cancelOuterCtx()
 
-	testMigrationVersionNum = setupTestSQLMigrations()
+	testMigrationVersionNum, sqlFs = setupTestSQLMigrations()
 
 	var err error
 	snapshotsBackupDir, err = os.MkdirTemp("", "snapshotbackup")
@@ -362,20 +363,20 @@ func TestMain(t *testing.M) {
 		log.Infof("%s", goldenSourceHistorySegment[4000].HistorySegmentID)
 		log.Infof("%s", goldenSourceHistorySegment[5000].HistorySegmentID)
 
-		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[1000].HistorySegmentID, "QmbogxbXpQsmkPhrsRGdoAMTeBuab2ALhPPHZA4hw8ifQ7", snapshots)
-		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[2000].HistorySegmentID, "QmWezSPFbRiY6u7WeohaafBWdCd5cfMiuFrVWUdSEZtyvJ", snapshots)
-		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[2500].HistorySegmentID, "QmNqyrM538DrNFmMsjAohwHSzPJNvyC64Q8n8hzEHvp6FJ", snapshots)
-		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[3000].HistorySegmentID, "QmcapMojGygbWJoNDcUeDyPALV9Lp2GefACqTjFqzLVvos", snapshots)
-		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[4000].HistorySegmentID, "QmNqy7u9fSCbZEUGk4EFTGkwFwJhaFnmRqxfjWorbo6mtH", snapshots)
-		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[5000].HistorySegmentID, "QmWUUBXB2AwHyinufVYNHKMQoA1ry1rpebsZzEbsJrjw29", snapshots)
-	}, postgresRuntimePath)
+		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[1000].HistorySegmentID, "QmS81XjevMhaw73DhUV7gzMRkm5LRMTQAtoYAxyTwrMP4d", snapshots)
+		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[2000].HistorySegmentID, "QmRefyDWqHBswf8jGhVjjHTDyL5y4u4ixUe2VhzkiBnxoT", snapshots)
+		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[2500].HistorySegmentID, "QmPNE3RtfrC1SYHcTd2mXbGx6Mcs6REB1rtpiXd5EH7sZf", snapshots)
+		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[3000].HistorySegmentID, "Qmeoe26PdeyKWYMdDzeQJtfDVtvzFi9vp8scmfnA4NRG1b", snapshots)
+		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[4000].HistorySegmentID, "QmaFYXoE41GBAn3Dz9z2WDewjxNMpB39BjamSmVdJrzHgn", snapshots)
+		panicIfHistorySegmentIdsNotEqual(goldenSourceHistorySegment[5000].HistorySegmentID, "QmZDfaXzDVpFZkZX31yMHxazQJ7UA3F5ymYXteocBUP1tk", snapshots)
+	}, postgresRuntimePath, sqlFs)
 
 	if exitCode != 0 {
 		log.Errorf("One or more tests failed, dumping postgres log:\n%s", postgresLog.String())
 	}
 }
 
-func setupTestSQLMigrations() int {
+func setupTestSQLMigrations() (int, fs.FS) {
 	sourceMigrationsDir, err := os.Getwd()
 	if err != nil {
 		panic(err)
@@ -387,6 +388,10 @@ func setupTestSQLMigrations() int {
 	testMigrationsDir, err = os.MkdirTemp("", "migrations")
 	if err != nil {
 		panic(err)
+	}
+
+	if os.Mkdir(filepath.Join(testMigrationsDir, sqlstore.SQLMigrationsDir), fs.ModePerm); err != nil {
+		panic(fmt.Errorf("failed to create migrations dir: %w", err))
 	}
 
 	var highestMigrationNumber int
@@ -412,7 +417,7 @@ func setupTestSQLMigrations() int {
 					return fmt.Errorf("failed to read file:%w", err)
 				}
 
-				err = os.WriteFile(filepath.Join(testMigrationsDir, info.Name()), data, fs.ModePerm)
+				err = os.WriteFile(filepath.Join(testMigrationsDir, sqlstore.SQLMigrationsDir, info.Name()), data, fs.ModePerm)
 				if err != nil {
 					return fmt.Errorf("failed to write file:%w", err)
 				}
@@ -432,13 +437,13 @@ func setupTestSQLMigrations() int {
 	}
 
 	testMigrationVersionNum := highestMigrationNumber + 1
-	err = os.WriteFile(filepath.Join(testMigrationsDir,
+	err = os.WriteFile(filepath.Join(testMigrationsDir, sqlstore.SQLMigrationsDir,
 		fmt.Sprintf("%04d_testmigration.sql", testMigrationVersionNum)), sql, fs.ModePerm)
 	if err != nil {
 		panic(err)
 	}
 
-	return testMigrationVersionNum
+	return testMigrationVersionNum, os.DirFS(testMigrationsDir)
 }
 
 func migrateDatabase(version int64) error {
@@ -512,7 +517,7 @@ func TestRestoringFromDifferentHeightsWithFullHistory(t *testing.T) {
 		assert.Equal(t, expectedBlocks, blocksFetched)
 		dehistoryService := setupDeHistoryService(ctx, log, inputSnapshotService, deHistoryStore, snapshotCopyFromPath, snapshotCopyToPath)
 
-		from, to, err := dehistoryService.LoadAllAvailableHistoryIntoDatanode(ctx)
+		from, to, err := dehistoryService.LoadAllAvailableHistoryIntoDatanode(ctx, sqlFs)
 		require.NoError(t, err)
 
 		assert.Equal(t, int64(1), from)
@@ -545,7 +550,7 @@ func TestRestoreFromPartialHistoryAndProcessEvents(t *testing.T) {
 
 	dehistoryService := setupDeHistoryService(ctx, log, inputSnapshotService, deHistoryStore, snapshotCopyFromPath, snapshotCopyToPath)
 
-	from, to, err := dehistoryService.LoadAllAvailableHistoryIntoDatanode(ctx)
+	from, to, err := dehistoryService.LoadAllAvailableHistoryIntoDatanode(ctx, sqlFs)
 	require.NoError(t, err)
 	assert.Equal(t, int64(2001), from)
 	assert.Equal(t, int64(3000), to)
@@ -635,7 +640,7 @@ func TestRestoreFromFullHistorySnapshotAndProcessEvents(t *testing.T) {
 
 	dehistoryService := setupDeHistoryService(ctx, log, inputSnapshotService, deHistoryStore, snapshotCopyFromPath, snapshotCopyToPath)
 
-	from, to, err := dehistoryService.LoadAllAvailableHistoryIntoDatanode(ctx)
+	from, to, err := dehistoryService.LoadAllAvailableHistoryIntoDatanode(ctx, sqlFs)
 	require.NoError(t, err)
 	assert.Equal(t, int64(1), from)
 	assert.Equal(t, int64(2000), to)
@@ -761,21 +766,29 @@ type sqlStoreBroker interface {
 func emptyDatabase() {
 	// NOTE: do not use databasetest.DeleteEverything(), for these we need a totally fresh database everytime to ensure
 	// we model as closely as possible what happens in practice
-
 	var err error
+	var poolConfig *pgxpool.Config
+
+	poolConfig, err = sqlConfig.ConnectionConfig.GetPoolConfig()
+	if err != nil {
+		panic(fmt.Errorf("failed to get pool config: %w", err))
+	}
+
+	db := stdlib.OpenDB(*poolConfig.ConnConfig)
+
+	if _, err = db.Exec(`SELECT alter_job(job_id, scheduled => false) FROM timescaledb_information.jobs WHERE proc_name = 'policy_refresh_continuous_aggregate'`); err != nil {
+		panic(fmt.Errorf("failed to stop continuous aggregates: %w", err))
+	}
+	db.Close()
+
 	for i := 0; i < 5; i++ {
-		err = sqlstore.RecreateVegaDatabase(context.Background(), logging.NewTestLogger(), sqlConfig.ConnectionConfig)
+		err = sqlstore.WipeDatabase(logging.NewTestLogger(), sqlConfig.ConnectionConfig, sqlFs)
 		if err == nil {
 			break
 		}
 		time.Sleep(5 * time.Second)
 	}
 
-	if err != nil {
-		panic(err)
-	}
-
-	err = sqlstore.CreateVegaSchema(logging.NewTestLogger(), sqlConfig.ConnectionConfig)
 	if err != nil {
 		panic(err)
 	}

--- a/datanode/dehistory/snapshot/orders/update_current_orders_test.go
+++ b/datanode/dehistory/snapshot/orders/update_current_orders_test.go
@@ -52,7 +52,7 @@ func TestMain(t *testing.M) {
 		sqlConfig = config
 		connectionSource = source
 		snapshotsDir = snapshotCopyToPath
-	}, postgresRuntimePath)
+	}, postgresRuntimePath, sqlstore.EmbedMigrations)
 }
 
 func TestUpdateCurrentOrdersState(t *testing.T) {

--- a/datanode/sqlstore/sqlstore.go
+++ b/datanode/sqlstore/sqlstore.go
@@ -18,6 +18,7 @@ import (
 	"embed"
 	"fmt"
 	"io"
+	"io/fs"
 	"time"
 
 	embeddedpostgres "github.com/fergusstrange/embedded-postgres"
@@ -57,8 +58,8 @@ func MigrateToLatestSchema(log *logging.Logger, config Config) error {
 	return nil
 }
 
-func MigrateToSchemaVersion(log *logging.Logger, config Config, version int64) error {
-	goose.SetBaseFS(EmbedMigrations)
+func MigrateToSchemaVersion(log *logging.Logger, config Config, version int64, fs fs.FS) error {
+	goose.SetBaseFS(fs)
 	goose.SetLogger(log.Named("db migration").GooseLogger())
 
 	poolConfig, err := config.ConnectionConfig.GetPoolConfig()
@@ -76,8 +77,8 @@ func MigrateToSchemaVersion(log *logging.Logger, config Config, version int64) e
 	return nil
 }
 
-func WipeDatabase(log *logging.Logger, config ConnectionConfig) error {
-	goose.SetBaseFS(EmbedMigrations)
+func WipeDatabase(log *logging.Logger, config ConnectionConfig, fs fs.FS) error {
+	goose.SetBaseFS(fs)
 	goose.SetLogger(log.Named("wipe database").GooseLogger())
 
 	poolConfig, err := config.GetPoolConfig()
@@ -94,7 +95,7 @@ func WipeDatabase(log *logging.Logger, config ConnectionConfig) error {
 	}
 
 	if currentVersion > 0 {
-		if err := goose.Down(db, SQLMigrationsDir); err != nil {
+		if err := goose.DownTo(db, SQLMigrationsDir, 0); err != nil {
 			return fmt.Errorf("failed to goose down the schema: %w", err)
 		}
 	}

--- a/datanode/sqlstore/sqlstore_test.go
+++ b/datanode/sqlstore/sqlstore_test.go
@@ -49,7 +49,7 @@ func TestMain(m *testing.M) {
 		testDBPort = cfg.ConnectionConfig.Port
 		testDBSocketDir = cfg.ConnectionConfig.SocketDir
 		connectionSource = source
-	}, postgresRuntimePath)
+	}, postgresRuntimePath, sqlstore.EmbedMigrations)
 }
 
 func DeleteEverything() {

--- a/datanode/utils/databasetest/setup.go
+++ b/datanode/utils/databasetest/setup.go
@@ -66,6 +66,7 @@ var (
 		"orders_live",
 		"parties",
 		"positions",
+		"positions_current",
 		"proposals",
 		"protocol_upgrade_proposals",
 		"ranking_scores",
@@ -85,7 +86,7 @@ var (
 )
 
 func TestMain(m *testing.M, onSetupComplete func(sqlstore.Config, *sqlstore.ConnectionSource, *bytes.Buffer),
-	postgresRuntimePath string,
+	postgresRuntimePath string, sqlFs fs.FS,
 ) int {
 	testDBSocketDir = filepath.Join(postgresRuntimePath)
 	testDBPort = 5432 // GetNextFreePort()
@@ -134,7 +135,7 @@ func TestMain(m *testing.M, onSetupComplete func(sqlstore.Config, *sqlstore.Conn
 		}
 		defer embeddedPostgres.Stop()
 
-		if err = sqlstore.WipeDatabase(log, sqlConfig.ConnectionConfig); err != nil {
+		if err = sqlstore.WipeDatabase(log, sqlConfig.ConnectionConfig, sqlFs); err != nil {
 			panic(err)
 		}
 

